src/graph_notebook/network/gremlin/GremlinNetwork.py

```diff
@@ -291,12 +302,28 @@ def add_vertex(self, v):
                     title += str(v[key])
                 label = title if len(title) <= self.label_max_length else title[:self.label_max_length - 3] + '...'
 
-            data = {'properties': properties, 'label': label, 'title': title}
+            # If the group_by_property is specified and it exists then assign that property to the group
+            # If group_by_property is not specified but the T.Label exists the assign it to the group
+            # If neither is true then do not assign group
+            for p in properties:
+                logger.debug(p)
+            if self.group_by_property:

```

Comment:
I think we can rework this now to make use of other suggestions. Something like this should work:

```
group = self.group_by_property if self.group_by_property in properties else ''
```

I think this would cover all three because the default value for `self.group_by_property` would now be T_LABEL. And if that default isn't in our properties, we fall back to an empty string. So we're effectively wrapping the first two conditions into one after amending the defaults.